Common/Securities/CashAccountMarginModel.cs

```diff
@@ -0,0 +1,242 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using QuantConnect.Orders;
+
+namespace QuantConnect.Securities
+{
+    /// <summary>
+    /// Provides an implementation of <see cref="ISecurityMarginModel"/> that requires cash of the correct
+    /// currency to exist in the portfolio to make trades. This also only applies to securities where shorting
+    /// is not an option. IOW, holdings.Quantity >= 0 at all times.
+    /// </summary>
+    public class CashAccountMarginModel : ISecurityMarginModel
+    {
+        private readonly CashBook _cashBook;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CashAccountMarginModel"/> class
+        /// </summary>
+        /// <param name="cashBook">The portfolio's cashbook</param>
+        public CashAccountMarginModel(CashBook cashBook)
+        {
+            _cashBook = cashBook;
+        }
+
+        /// <summary>
+        /// Cash accounts require the full value of the order plus any fees
+        /// </summary>
+        /// <param name="security"></param>
+        /// <param name="order">The order to be executed</param>
+        /// <returns>The total margin in terms of the currency quoted in the order</returns>
+        public decimal GetInitialMarginRequiredForOrder(Security security, Order order)
+        {
+            var fees = security.FeeModel.GetOrderFee(security, order);
+            var marginRequiredInAccountCurrency = order.GetValue(security) + fees;
```

Comment:
nit -- terminology here is incorrect, perhaps we should be phrasing it as `cashRequired` or similar